### DISCUSSION TITLE
Mac M1 use of podman

### DISCUSSION THREAD

**Discussion by bobwolff68** [containers/podman#13433] [Q&A]
New to podman - using a Macbook Air M1 machine as I understand podman supports this now - great! But my experience is that I cannot get `podman machine start` to work. It's unclear to me if I can only use a subset of machine types due to the x86/arm64 divide or what, but the general instructions didn't show me much love - see below. Any thoughts?

`➜  ~ podman --version`
`podman version 3.4.4`
`➜  ~ podman machine list`
`NAME                     VM TYPE     CREATED       LAST UP       CPUS        MEMORY      DISK SIZE`
`podman-machine-default*  qemu        28 hours ago  28 hours ago  1           2.147GB     10.74GB`
`➜  ~ podman machine start`
`INFO[0000] waiting for clients...`
`INFO[0000] listening tcp://127.0.0.1:7777`
`INFO[0000] new connection from  to /var/folders/b2/nchtpdq54n7fbjmwkjw1c6xh0000gp/T/podman/qemu_podman-machine-default.sock`
`Waiting for VM ...`
`qemu-system-x86_64: Unknown Error`
`Error: dial unix /var/folders/b2/nchtpdq54n7fbjmwkjw1c6xh0000gp/T/podman/podman-machine-default_ready.sock: connect: connection refused`
`ERRO[0003] cannot receive packets from , disconnecting: cannot read size from socket: EOF`
`ERRO[0003] cannot read size from socket: EOF`

**Comment by afbjorklund** [comment]
It seems like you used the wrong binary for podman-remote, if `podman` is amd64 then it will also try to emulate an amd64 VM. (oops)

Can you make sure that your `podman` binary is arm64 ? Then it should be running `qemu-system-aarch64`, **not** `qemu-system-x86_64`.

You also want to install the qemu-user-static binaries, so that you can run containers from the other architecture - once the VM is up.

* https://github.com/containers/podman/issues/12144

**Reply by bobwolff68** [reply]
Thanks for the reply and insight!

I think my output above confirms that it is running `qemu-system-x86_64` given the `: Unknown Error`.

I'm happy to install the qemu-user-static binaries, etc, but the referenced issue 12144 doesn't make that clear to me. Can you help me get clear on what to do?

And is there something that I did incorrectly in installation? podman says it supports M1 so it seems I shouldn't have to be using a crowbar to get it rolling. :)

**Reply by afbjorklund** [reply]
Homebrew installs in `/opt/homebrew` for arm64, and `/usr/local` for amd64.
(On the Mac that is, it installs in `/home/linuxbrew/.linuxbrew` on Linux...)

See https://docs.brew.sh/Installation

To enable running amd64 containers, in an arm64 virtual machine (e.g. M1):
(Native containers, or multi-arch, work out-of-the-box, this was for emulation)

`podman machine ssh sudo rpm-ostree install qemu-user-static`
`podman machine ssh sudo systemctl reboot`

**Comment by baude** [comment]
there is some cases where is brew was being run with rosettta, the arch is being set to x86_64.  This is not a Podman thing.  I think some people has success uninstalling and re-installed brew.

**Reply by afbjorklund** [reply]
Well, it's a bug that podman machine doesn't work in emulation - on the other hand, that's normally **not** what the user wants anyway.

* https://github.com/containers/podman/issues/11458

But should be able to do better than "Unknown Error", no ?

* https://github.com/containers/podman/issues/13010
